libs/editor/editor_notes.cpp

```diff
@@ -155,12 +153,21 @@ void Notes::CreateNote(ms::LatLon const & latLon, std::string const & text)
 
 void Notes::Upload(osm::OsmOAuth const & auth)
 {
+  if (m_isUploadingNow)
+    return;
+  m_isUploadingNow = true;
+
   // Capture self to keep it from destruction until this thread is done.
-  auto const self = shared_from_this();
+  /// @todo It looks redundant because of parent static Editor instance.
+  auto self = shared_from_this();
```

Comment:
- It is redundant, because m_notes is a member of a static editor instance
- It was made as a shared pointer either by mistake, or because of the editor tests which need to reset the state in ResetNotes()

Refactor?